examples/weight_compression/openvino/tiny_llama/main.py

```diff
@@ -0,0 +1,211 @@
+            group_size=group_size,
+            all_layers=True,
+        )
+        all_layers_similarity = evaluate_fn(optimized_model=full_optimized_model)
+        if all_layers_similarity >= 1 - MAX_DROP:
+            print_results(full_optimized_model, ratio, group_size, all_layers_similarity)
+        else:
+            print_results(optimized_model, ratio, group_size, similarity)
+        return
+
+    ratio, group_size = param_grid[-1]  # (MIN_GROUP_SIZE, MIN_RATIO)
+    optimized_model = compress_model(original_ov_model, nncf_dataset, ratio, group_size)
+    similarity = evaluate_fn(optimized_model=optimized_model)
+    if similarity < 1 - MAX_DROP:
+        nncf_logger.info(
+            "The model was compressed with the minimum ratio and group_size, ",
+            "but it could not achieve the required accuracy drop. ",
+            "We recommend choosing a different mode for weight compression."
+        )
+        print_results(optimized_model, ratio, group_size, similarity)
+        return
+
+    for ratio, group_size in param_grid[1:-1]:
+        optimized_model = compress_model(original_ov_model, nncf_dataset, ratio, group_size)
+        similarity = evaluate_fn(optimized_model=optimized_model)
+        if similarity >= 1 - MAX_DROP:
+            print_results(optimized_model, ratio, group_size, similarity)
+            return
+
+    optimized_model = compress_model(original_ov_model, nncf_dataset, MIN_RATIO, MIN_GROUP_SIZE)
+    print_results(optimized_model, MIN_RATIO, MIN_GROUP_SIZE, similarity)
+
+
+def tiny_llama_transform_func(item, tokenizer):
+    tokens = tokenizer(item["text"])
+    attention_mask = np.expand_dims(np.array(tokens["attention_mask"]), 0)
+    position_ids = attention_mask.cumsum(-1) - 1
+    position_ids = np.ma.array(position_ids, mask=attention_mask == 0)
+    position_ids.filled(fill_value=1)
+    res = {
+        "input_ids": np.expand_dims(np.array(tokens["input_ids"]), 0),
+        "attention_mask": attention_mask,
+        "position_ids": position_ids,
+    }
+
+    def gen_pkv(num_heads, head_dim, num_layers):
+        res = {}
+        for i in range(num_layers):
+            res[f"past_key_values.{i}.key"] = np.zeros((1, num_heads, 0, head_dim))
+            res[f"past_key_values.{i}.value"] = np.zeros((1, num_heads, 0, head_dim))
+        return res
+
+    res.update(gen_pkv(4, 64, 22))
+    return res
+
+
+model_id = "PY007/TinyLlama-1.1B-step-50K-105b"
+ov_config = {"PERFORMANCE_HINT": "LATENCY", "NUM_STREAMS": "1", "CACHE_DIR": ""}
+model = OVModelForCausalLM.from_pretrained(
+    model_id,
+    export=True,
+    trust_remote_code=True,
+    use_cache=True,
+    ov_config=ov_config,
+)
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+dataset = load_dataset("wikitext", "wikitext-2-v1", split="train[:1000]")
+dataset = dataset.filter(lambda example: len(example["text"]) > 128)
+transform_func = partial(tiny_llama_transform_func, tokenizer=tokenizer)
+
+start = datetime.datetime.now()
```

Comment:
What is the usual time required to run this sample?

Reply:
The example requires 45 minutes